src/test/java/com/nokia/uwr/connectionmanager/ConnectionManagerImplTest.java

```diff
@@ -0,0 +1,58 @@
+package com.nokia.uwr.connectionmanager;
+
+import com.nokia.uwr.model.BTS;
+import com.nokia.uwr.model.UEMeasurement;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class ConnectionManagerImplTest {
+
+    private ConnectionManager connectionManager;
+    private BTS expectedBTS;
+
+    @BeforeEach
+    public void setup() {
+        connectionManager = new ConnectionManagerImpl();
+
+        expectedBTS = new BTS("BTS1", 50);
+
+        Map<BTS, List<UEMeasurement>> assignments = new HashMap<>(
```

Comment:
What is the reason for converting immutable map into mutable map?

Reply:
Can you point the place in this test class where the map is being mutated?

Reply:
Not in test class, but by design it should be mutable, so what's the point of making it immutable just because we don't just modify it in this class?